agenta-backend/agenta_backend/models/db_models.py

```diff
@@ -6,6 +6,20 @@
 from odmantic import EmbeddedModel, Field, Model, Reference
 
 
+class APIKeyDB(Model):
+    prefix: str
+    hashed_key: str
+    user_id: str
+    rate_limit: int = Field(default=1000)
+    expiration_date: Optional[datetime]
+    blacklist: bool = Field(default=False)
```

Comment:
blacklist !? like why ? why do we need this

Reply:
Yeah, we may not exactly need it now. 

We won't if users are allowed to fully manage their api keys, and they can delete it instead of blacklisting it.